component/src/main/java/org/cbioportal/genome_nexus/util/GenomicVariantUtil.java

```diff
@@ -138,6 +138,42 @@ public static boolean isMafFile(String file) {
         return Pattern.matches("^[\\w\\-\\./]+(\\.maf)|^[\\w\\-\\./]+(\\.tsv)|^[\\w\\-\\./]+(\\.txt)$$", file);
     }
 
+    // Hgvs formatted variants of several types are parsed for provided Reference Alleles
+    // empty string is returned if no reference allele can be extracted
+    public static String providedReferenceAlleleFromHgvs(String hgvs) {
+        // Substitutions require a single nucleotide reference sequence:
+        // 1g.123456A>G  (chromosome 1, genomic position 123456, reference sequence A, substituted with G)
+        // ref: http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/
+        String match = getPattern("(?<=[^ATGC])[ATGC]>", hgvs);
```

Comment:
I'm not sure how often such requests are received by the endpoints at genomenexus.org. The import pipeline uses the genomic endpoints, and the mutations tab of the query response page in cbioportal also makes queries using genomicLocation objects posted to "https://v1.genomenexus.org/annotation/genomic". So it may be that normal operations with portal imports and websites do not use the /annotation endpoint at all. However, the genome nexus server code was recently changed so that when sending requests to Ensembl's VEP webapi using HGVS, we standardize our requests to a format which omits the reference allele string --- and this seems to be the recommended standard now here : https://varnomen.hgvs.org/ so maybe we should expect/support the (simplified) format.